minecraft/networking/types.py

```diff
@@ -213,3 +223,31 @@ def read(file_object):
     @staticmethod
     def send(value, socket):
         socket.send(uuid.UUID(value).bytes)
+
+
+class Position(Type):
+    @staticmethod
+    def read(file_object):
+        location = UnsignedLong.read(file_object)
+        x = int(location >> 38)
+        y = int((location >> 26) & 0xFFF)
+        z = int(location & 0x3FFFFFF)
+
+        if x >= pow(2, 25):
+            x -= pow(2, 26)
+
+        if y >= pow(2, 11):
+            y -= pow(2, 12)
+
+        if z >= pow(2, 25):
+            z -= pow(2, 26)
+
+        return {'x': x, 'y': y, 'z': z}
+
+    @staticmethod
+    def send(value, socket):
```

Comment:
I would suggest changing this to pass either the x, y, z as arguments or passing a dict containing x, y, z. The types classes are supposed to abstract away the python types and sending them over the network and calling an encode method before sending breaks that a little.